netsim/modules/vlan.py

```diff
@@ -391,6 +389,77 @@ def set_node_vlan_mode(n_vlan: Box, node: Box, g_vlan: Box, topology: Box) -> No
                 g_vlan.get('mode',None) or \
                 topology.get('vlan.mode',None) or 'irb'
 
+"""
+populate_node_vlan: merge node-level and topology-level VLAN data
+
+* We're assuming the VLAN needs to be created, so we're not tiptoeing around.
+* Node VLAN mode is evaluated using the precedence principles, everything else
+  is merged
+* _global_merge flag is set once we're done so we're not doing the same thing
+  too many times
+"""
+def populate_node_vlan(vlan: str, node: Box, topology: Box) -> Box:
+  if '_global_merge' in node.vlans[vlan]:
+    return node.vlans[vlan]
```

Comment:
Does this happen?

Reply:
Yes, we call that function several times, first in `pre_transform` hook to build the node VLANs configured on nodes or groups, and later when processing access/trunk links to ensure the VLAN is also defined on the node.